replications/internal/queue_management.go

```diff
@@ -11,18 +12,23 @@ import (
 	"github.com/influxdata/influxdb/v2/kit/platform"
 	"github.com/influxdata/influxdb/v2/pkg/durablequeue"
 	"github.com/influxdata/influxdb/v2/replications/metrics"
+	"github.com/influxdata/influxdb/v2/replications/remotewrite"
 	"go.uber.org/zap"
 )
 
+type remoteWriter interface {
+	Write(context.Context, []byte) error
+}
+
 type replicationQueue struct {
-	id        platform.ID
-	queue     *durablequeue.Queue
-	wg        sync.WaitGroup
-	done      chan struct{}
-	receive   chan struct{}
-	logger    *zap.Logger
-	metrics   *metrics.ReplicationsMetrics
-	writeFunc func([]byte) error
+	id           platform.ID
+	queue        *durablequeue.Queue
+	wg           sync.WaitGroup
+	done         chan struct{}
+	receive      chan struct{}
+	logger       *zap.Logger
+	metrics      *metrics.ReplicationsMetrics
+	remoteWriter remoteWriter
```

Comment:
The writer capability ended up being more complicated than I could fit into a list of function parameters, so this changed to a separate struct.